File: ruby-exercises/sum_all.rb
require 'rspec'


def sum_all num1, num2
  
  if Integer === num1 && Integer === num2
    if num1 < 0 || num2 < 0 
      'ERROR'
    elsif num1 > num2
      range = (num2..num1).to_a
      range.reduce(:+)
    else
      range = (num1..num2).to_a
      range.reduce(:+)
    end
  else
    'ERROR'
  end

end


describe 'sum_all' do 
  it 'sums numbers within the range' do 
    expect(sum_all(1, 4)).to eq(10)
  end

  it 'works with large numbers' do 
    expect(sum_all(1, 4000)).to eq(8002000)
  end

  it 'works with larger number first' do 
    expect(sum_all(123, 1)).to eq(7626)
  end

  it 'returns ERROR with negative numbers' do 
    expect(sum_all(-10, 4)).to eq('ERROR')
  end

  it 'returns ERROR with non-number parameters' do 
    expect(sum_all(10, '90')).to eq('ERROR')
  end

  it 'returns ERROR with non-number parameters' do 
    expect(sum_all(10, [90, 1])).to eq('ERROR')
  end
end

File: python-exercises/sum_all/sum_all.py

def sum_all(num1, num2):
  if type(num1) != int or type(num2) != int:
    return 'ERROR'
  elif num1 < 0 or num2 < 0:
    return 'ERROR'
  elif num1 > num2:
    my_range = list(range(num2, num1 + 1))
    total = 0
    for num in my_range:
      total = total + num 
    return total
  else:
    my_range = list(range(num1, num2 + 1))
    total = 0
    for num in my_range:
      total = total + num 
    return total
